src/js/src/mixins/SearchUtils.js

```diff
@@ -101,6 +101,61 @@ export default {
       else {
         this.deliverSearchRequest(futureQuery, updateHistory, pagnation)
       }
-    }
+    },
+
+    /*
+    * Regex query checker
+    * Kudos to tokee
+    * https://github.com/netarchivesuite/solrwayback/issues/41#issuecomment-724571414
+    * https://jsfiddle.net/bsc6zkxy/12/   
+    **/  
+    $_checkQueryForBadSyntax(q) {
+        let responses = []
```

Comment:
A simple `q = q.replace('\n', ' ');` should ensure that multi-line queries are also checked for syntactic problems.

Reply:
Codescanner seems to have a point:
Check warning on line 113 in src/js/src/mixins/SearchUtils.js

Code scanning
Incomplete string escaping or encoding
This replaces only the first occurrence of '\n'.

If i make 2 line breaks and do bad stuff on the last line, its not caught.

So this: 

mus fisk AND
test AND test
and test

Is not caught at the moment, at least not when I tried it.

Reply:
Nice catch! If should be `q = q.replaceAll("\n", "*");` (or we could switch to regexp with `q = q.replace(/\n/g, "*");` as the `/g`-modifier only works with regexp, but since this is simple single-char replacement, there is no need for the switch).

Reply:
Thank you Code scanner! I agree replaceAll would be the best and simplest path here. But why switch to replacing with "*" it was ' ' before @tokee?

Reply:
Dóh! That was just me testing in a way where I could see the result. It should of course be ` `. Sorry.